src/hera/workflows/task.py

```diff
@@ -145,6 +147,46 @@ def finished_at(self) -> str:
     def result(self) -> str:
         return f"{{{{tasks.{self.name}.outputs.result}}}}"
 
+    def get_parameters_as(self, name):
+        """Gets all the output parameters from this task"""
```

Comment:
@flaviuvadan couple of suggestions - can we instead introduce get_output? which gets an appropriate artifact or parameter?

The other thing we noticed is that `valueFrom` + expression is a better way to reference variables rather than using the substitution syntax. The latter doesn't work for inline template outputs.

Artifact similarly has a fromExpression field.

Reply:
> can we instead introduce get_output? which gets an appropriate artifact or parameter?

Can one have the same name for a parameter and an artifact? If that is the case then `get_parameters` seems more precise 🤔 

> The other thing we noticed is that valueFrom + expression is a better way to reference variables rather than using the substitution syntax. The latter doesn't work for inline template outputs.

> Artifact similarly has a fromExpression field

I am not sure I fully grasp this - do you mind telling me more? My hunch is that you are referring to [this field](https://github.com/argoproj-labs/hera-workflows/blob/main/src/hera/workflows/artifact.py#L36) and implying that building an expression via `hera.expr` is better, perhaps?

Reply:
For the latter,

valueFrom: expression: tasks.name.output.result

works everywhere but

value: {{tasks.name.output.result}} doesn't

Reply:
Oh, ok, got it